pkg/utils/os/env_test.go

```diff
@@ -0,0 +1,50 @@
+package os
+
+import (
+	"os"
+	"testing"
+)
+
+func TestGetEnvWithFallback(t *testing.T) {
+	type args struct {
+		name     string
+		fallback string
+	}
+	tests := []struct {
+		name  string
+		setup map[string]string
+		args  args
+		want  string
+	}{
+		{
+			name: "fallback",
+			args: args{
+				"xxx",
+				"yyy",
+			},
+			want: "yyy",
+		},
+		{
+			name: "no fallback",
+			setup: map[string]string{
+				"xxx": "zzz",
+			},
+			args: args{
+				"xxx",
+				"yyy",
+			},
+			want: "zzz",
+		},
+	}
+	for _, tt := range tests {
+		os.Clearenv()
+		t.Run(tt.name, func(t *testing.T) {
+			for k, v := range tt.setup {
+				os.Setenv(k, v)
```

Comment:
should we use t.Setenv instead so that it automatically cleans it up post tests?

Reply:
good catch, fixed